src/core/ext/filters/client_channel/client_channel.c

```diff
@@ -1210,6 +1210,7 @@ static void pick_callback_cancel_locked(grpc_exec_ctx *exec_ctx,
   }
   grpc_lb_policy_cancel_pick_locked(exec_ctx, calld->lb_policy,
                                     &calld->connected_subchannel, error);
+  GRPC_LB_POLICY_UNREF(exec_ctx, calld->lb_policy, "pick_cancelled");
```

Comment:
I don't think this is right.  Cancelling the pick will cause the pick callback (`pick_callback_done_locked()`) to be invoked immediately with an error, and that's where the unref will be done.  If we do it here as well, then we'll unref too many times.